### PR TITLE
feat(ocb3): implement flexible nonce size

### PR DESCRIPTION
**Pull request by dignifiedquire**
Tested with nonce size `15` in rpgp. Unfortunately I don't have any additional test vectors.

### CODE REVIEW

File: ocb3/src/lib.rs [hunk 1]
```diff
@@ -77,7 +77,21 @@ mod private {
...
+    impl SealedNonceSize for consts::U6 {}
+    impl SealedNonceSize for consts::U7 {}
+    impl SealedNonceSize for consts::U8 {}
+    impl SealedNonceSize for consts::U9 {}
+    impl SealedNonceSize for consts::U10 {}
+    impl SealedNonceSize for consts::U11 {}
     impl SealedNonceSize for consts::U12 {}
+    impl SealedNonceSize for consts::U13 {}
+    impl SealedNonceSize for consts::U14 {}
+    impl SealedNonceSize for consts::U15 {}
```
Comment (tarcieri) [inline ocb3/src/lib.rs:94]: Hmm, should we change this to an [`LeEq`](https://docs.rs/typenum/latest/typenum/operator_aliases/type.LeEq.html) constraint instead?

Comment (tarcieri) [reply ocb3/src/lib.rs:94]: The same probably goes for `TagSize`

Comment (dignifiedquire) [reply ocb3/src/lib.rs:94]: yes that seems better

Comment (dignifiedquire) [reply ocb3/src/lib.rs:94]: okay, I have no idea how to use this though to make a blanket impl 😅

Comment (tarcieri) [reply ocb3/src/lib.rs:94]: I opened #590 to use `IsLessOrEqual` for both `NonceSize` and `TagSize`

Comment (dignifiedquire) [reply ocb3/src/lib.rs]: rebased accordingly

**Review by tarcieri — APPROVED** [review]

**Pull request merged**